### PR TITLE
Deprecate wiki

### LINKED ISSUE #2346: Deprecate wiki
## Change
As briefly talked about in the #daala-weekly yesterday, the wiki is quite outdated and should be deprecated.

The question is: **Which of the pages should be kept and moved to the `doc/` folder?** The others will just be deleted.

## Pages
| file                          | purpose                                              | author(s)         | action |
| :---------------------------- | :--------------------------------------------------- | :---------------- | :----- |
| `Frame-Type-Selection.md`     | Explains frame type selection                        | shssoichiro       | pick   |
| `Home.md`                     | Short hello message                                  | mbebenita         | drop   |
| `Quality-&-Speed-Features.md` | Various features compared to compeititors (outdated) | mbebenita, rzumer |        |
| `Roadmap.md`                  | Feature plan for AV1 Encoder                         | mbebenita, negge  |  drop  |

## Procedure
It's obvious that `Home.md` can be dropped, but for the others I would just like consult the authors (@shssoichiro, @mbebenita, @rzumer, @negge).

Please state if the page should be kept, and if yes, if it needs to be updated :)
Thank you! :smile:

### PR DESCRIPTION
**Pull request by Urhengulas**
# Description

This PR deprecates the wiki, by moving or dropping the files to the main repo.

Additionally the documentation in the repo gets a cleanup. Also it gets tweaked for potential [GitHub-Pages](https://pages.github.com/) Support (example: https://urhengulas.github.io/rav1e)

Fixes #2346 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change which doesn't change functionality
- [x] Documentation update

## Open TODOs
- [ ] Handle `Quality-&-Speed-Features.md` (waiting for feedback)
- [x] Handle `Roadmap.md` -> drop it + @negge will go through it and convert potential open questions to issues

# How Has This Been Tested?

- [x] `README.md` and `docs/*` look good and render correctly
- [x] Looks good on gh-pages

### CODE REVIEW

**Comment by lu-zero** [comment]
I'd rather squash those commit keeping the co-author tags.

@Luni-4 can you please discuss with @Urhengulas regarding setting up [mdbook](https://github.com/rust-lang/mdBook) to render everything? :)

**Comment by Urhengulas** [comment]
> Please squash relevant changes, having 38 commits for the patchset seems a bit odd.

I wanted to preserve the history of the original wiki for future reference, but probably that's not needed so I will just squash it.

**Comment by vibhoothiiaanand** [comment]
Also, 
Having pages generate with GitHub Pages inside rav1e seems not so proper solution to me since we may move our infra to GitLab or something else in future which is not sure. So, in that case, it may become problematic since we are relying on GitHub Pages for the generation. It would be cool to have a generic solution like `mdbook`, or having it within the main website etc.

**Comment by Urhengulas** [comment]
@vibhoothiiaanand 

---

> Having pages generate with GitHub Pages inside rav1e seems not so proper solution to me since we may move our infra to GitLab or something else in future which is not sure.

Okay. `mdbook` sounds like a good vendor-independent solution. I will look into this together with @Luni-4 in a separate PR.

---
> Is it necessary to rename the folder to `docs` from `doc`

This is only "necessary" for gh-pages. If we say we don't want to have that anyways, it is not needed.

**Comment by Luni-4** [comment]
> @Luni-4 can you please discuss with @Urhengulas regarding setting up [mdbook](https://github.com/rust-lang/mdBook) to render everything? :)

Sure! 

> Okay. `mdbook` sounds like a good vendor-independent solution. I will look into this together with @Luni-4 in a separate PR.

@Urhengulas Feel free to contact me on IRC if you need some help or in your future PR, np for me :)

File: Quality-&-Speed-Features.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+PSA: this page is extremely outdated. Please refer to `README.md` and the Rust documentation for the `config` module.
+
+### Required Quality Features
+
+Quality features that are required for to reach parity with state-of-the-art encoders.
+
+| Feature       |  rav1e        | libaom | x264 | x265 |
+| ------------- |:-------------:|:-----:|:-----:|:---:|
+| Exact RDO |:white_check_mark:| :x: |:white_check_mark:|:white_check_mark:|
+| Motion-adaptive quantizer (CRF) |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:10]: drop this line

File: Quality-&-Speed-Features.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+| Temporal RDO |:white_check_mark:|:x:|:white_check_mark: (mbtree) |:white_check_mark:|
+| Smart distortion |:white_check_mark:| Broken (cdef-dist, daala-dist) |:white_check_mark: (psyrd) |:white_check_mark:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:12]: X instead of Broken

File: Quality-&-Speed-Features.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+| Automatic QM |:white_check_mark:|:x:|:x:|:white_check_mark:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:13]: X, we don't have this

File: Quality-&-Speed-Features.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+| Rational luma/chroma weight |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
+| B-pyramid |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:15]: libaom does have this now

File: Quality-&-Speed-Features.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+|  Good rate control |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:16]: drop this line

File: Quality-&-Speed-Features.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+### Unique Quality Features
+
+Quality features where rav1e can go beyond state-of-the-art encoders.
+
+| Feature       |  rav1e        | libaom | x264 | x265 |
+| ------------- |:-------------:|:-----:|:-----:|:---:|
+| Joint loop filter search |:white_check_mark:|:x:|:x:|:x:|
+| Chunk-compatible first pass |:white_check_mark:|:x:|:x:|:x: (in UHDKit) |
+| Dynamic programming mvs |:white_check_mark:|:x:|:x:|:x:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:26]: drop this line

File: Quality-&-Speed-Features.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+| Auto film grain |:white_check_mark:|:x:|:x:|:x:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:27]: drop this line

File: Quality-&-Speed-Features.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+### Speed Features
+
+| Feature       |  rav1e        | libaom | x264 | x265 |
+| ------------- |:-------------:|:-----:|:-----:|:---:|
+| Pruning using approximate RDO |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:33]: libaom has this

File: Quality-&-Speed-Features.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+| Frame-parallel encoding |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:34]: rav1e does not have this

File: Quality-&-Speed-Features.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+| ML trained pruning |:white_check_mark:|:x:|:x:|:x:|
+| Approximate transforms |:white_check_mark:|:white_check_mark: (by accident) |:x:|:x:|
+| Model-based RDO cost |:white_check_mark:|:x: (broken) |:white_check_mark:| ? |
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:37]: rav1e does not have this

File: Quality-&-Speed-Features.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+| Auto film grain |:white_check_mark:|:x:|:x:|:x:|
+
+### Speed Features
+
+| Feature       |  rav1e        | libaom | x264 | x265 |
+| ------------- |:-------------:|:-----:|:-----:|:---:|
+| Pruning using approximate RDO |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
+| Frame-parallel encoding |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
+| ML trained pruning |:white_check_mark:|:x:|:x:|:x:|
+| Approximate transforms |:white_check_mark:|:white_check_mark: (by accident) |:x:|:x:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:36]: drop this

File: Quality-&-Speed-Features.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+| Dynamic programming mvs |:white_check_mark:|:x:|:x:|:x:|
+| Auto film grain |:white_check_mark:|:x:|:x:|:x:|
+
+### Speed Features
+
+| Feature       |  rav1e        | libaom | x264 | x265 |
+| ------------- |:-------------:|:-----:|:-----:|:---:|
+| Pruning using approximate RDO |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
+| Frame-parallel encoding |:white_check_mark:|:x:|:white_check_mark:|:white_check_mark:|
+| ML trained pruning |:white_check_mark:|:x:|:x:|:x:|
```
Comment (tdaede) [inline Quality-&-Speed-Features.md:35]: drop this

**Comment by Urhengulas** [comment]
Thanks @tdaede for the input. I addressed everything.

I think the PR should be ready now.

File: README.md [hunk 13]
```diff
@@ -20,13 +40,23 @@ rav1e is an AV1 video encoder. It is designed to eventually cover all use cases,
 * Constant quantizer and target bitrate (single- and multi-pass) encoding modes
 * Still picture mode
 
-## Releases
+## Documentation
+Find the documentation in [`doc/README.md`](doc/README.md)
+* AOM_ANALYZER
```
Comment (lu-zero) [inline README.md:45]: What is this list about?

File: README.md [hunk 14]
```diff
@@ -97,61 +122,41 @@ cargo install cargo-c
...
+rav1e has several optional features that can be enabled by passing `--features` to cargo. Passing `--all-features` is discouraged.
 
-rav1e has several optional features that can be enabled by passing --features to cargo test. Passing --all-features is discouraged.
+#### Features
+Find a full list in feature-table in [`Cargo.toml`](Cargo.toml)
 
-* asm - enabled by default. When enabled, assembly is built for the platforms supporting it.
+* `asm` - enabled by default. When enabled, assembly is built for the platforms supporting it.
   * It requires `nasm` on `x86_64`.
   * It requires `gas` on `aarch64`.
```
Comment (lu-zero) [inline README.md:153]: While at it might be good to mention that the clang assembler is also supported.

File: doc/AOM_ANALYZER.md [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+<summary><b>Table of Content</b></summary>
+
+- [Local Analyzer](#local-analyzer)
+- [Online Analyzer](#online-analyzer)
+</details>
+
+## Local Analyzer
+
+1. Download the [AOM Analyzer](http://aomanalyzer.org) ([source](https://github.com/xiph/aomanalyzer)).
+2. Download [inspect.js](https://people.xiph.org/~mbebenita/analyzer/inspect.js) and [inspect.wasm](https://people.xiph.org/~mbebenita/analyzer/inspect.wasm) and save them in the same directory.
```
Comment (lu-zero) [inline doc/AOM_ANALYZER.md:13]: Those files aren't available, please point to the correct place (@tdaede which is it?)

File: README.md (hunk 13 above)
Comment (Urhengulas) [reply README.md:45]: This is table of contents of the documentation. I thought it is helpful for people to know what they can find there.

File: doc/AOM_ANALYZER.md (hunk 15 above)
Comment (Urhengulas) [reply doc/AOM_ANALYZER.md:13]: Found them: https://github.com/xiph/aomanalyzer/issues/80

File: README.md (hunk 13 above)
Comment (lu-zero) [reply README.md:45]: if you make links to the files it might help a bit, as it is is quite confusing.

File: README.md (hunk 14 above)
Comment (Urhengulas) [reply README.md:153]: Can you maybe add the instructions how this works?

File: README.md (hunk 13 above)
Comment (Urhengulas) [reply README.md:45]: If you think it is not helpful I will remove it to keep it compact.

File: README.md (hunk 14 above)
Comment (Urhengulas) [reply README.md]: Addressed in https://github.com/xiph/rav1e/pull/2365/commits/b3de4e60a5c5562d5a9b8ecd2da8d3f0ad87b271

**Comment by Urhengulas** [comment]
Thank you everyone for keeping patient :)
I squashed the set a bit to be more compact

**Pull request merged**